### PR TITLE
Add public.truetype.overlap and public.truetype.compositeFlags

### LINKED ISSUE #162: TrueType flags in UFO
Taken from [TrueType instructions in UFO #93](https://github.com/unified-font-object/ufo-spec/issues/93).

We need to address the composite glyph component flags (overlap, round, useMyMetrics) mentioned in https://github.com/unified-font-object/ufo-spec/issues/93#issuecomment-650106369 or https://github.com/unified-font-object/ufo-spec/issues/93#issuecomment-650253676.

From the discussion here is what it could look like:
```xml
<key>public.objectsLib</key>
<dict>
  <key>bANWxQTe5a</key>
  <dict>public.truetype.flags</dict>
    <key>useMyMetrics</key>
    <true/>
    <key>overlap</key>
    <false/>
    <key>round</key>
    <true/>
  </dict>
</dict>
```

Technically the useMyMetrics flag should only be used for one component. The OT spec doesn’t explicitly restrict it to one component but it is understood that only the sidebearing of only one component can be used.

The overlap flag must be only set for the first component, this is explicit in the OT spec.
This is more specific to the glyph than to the component.
There is also a OVERLAP_SIMPLE flag for contour glyphs, it also applies explicitly only to the first flag byte of the glyph.
We should probably have a single flag in UFO that can be use for the composite glyph component OVERLAP_COMPOUND flag or the simple contour glyph OVERLAP_SIMPLE.
Should we have a `glif.lib["public.truetype.overlap"]` key instead of a component.lib key through objectsLib?

### PR DESCRIPTION
**Pull request by moyogo**
Fixes https://github.com/unified-font-object/ufo-spec/issues/162.

This adds a glyph lib key `public.truetype.overlap` as a boolean and the component object-level lib keys `public.truetype.useMyMetrics` and `public.truetype.roundOffsetToGrid` as boolean as well.

The overlap can be used to set the TrueType Simple Glyph flags bit 6 (OVERLAP_SIMPLE) or the Component Glyph flags bit 10 (OVERLAP_COMPOUND).

The roundOffsetToGrid can be used to set the Component Glyph flags bit 2 (ROUND_XY_TO_GRID).

The useMyMetrics can be used to set the Component Glyph flags bit 9 (USE_MY_METRICS). In theory tools may use this to update the xOffset and yOffset of a component and the width, as well as height and verticalOrigin, of the composite glyph that contains it.

### CODE REVIEW

**Comment by benkiel** [comment]
This LGTM, but would like a second opinion before merging.

**Review by justvanrossum — APPROVED** [review]

File: versions/ufo3/glyphs/glif.md
Comment (anthrotype) [inline versions/ufo3/glyphs/glif.md]:
```suggestion
| public.truetype.roundOffsetToGrid | boolean | This key is used in a component object-level dictionary and, when set to true, indicates bit 2 (ROUND_XY_TO_GRID) listed in the [OpenType glyf Composite Glyph Description] should be set for the component in the Component Glyph flags in TrueType binaries. Authoring tools may ignore this data. This data is optional.
```

File: versions/ufo3/glyphs/glif.md
Comment (anthrotype) [inline versions/ufo3/glyphs/glif.md]: I don't quite understand "It may be used to update the `xOffset, yOffset` of the component and the `width`, `height` and `verticalOrigin` of the glyph it is in." Do we actually need to try explain what "useMyMetrics" does, can't we simply refer to the truetype spec for its meaning?

**Review by anthrotype — APPROVED** [review]

File: versions/ufo3/glyphs/glif.md
Comment (moyogo) [reply versions/ufo3/glyphs/glif.md]: That can be removed. I think that makes sense for a more general `public.useMyMetrics` but let's keep this as `public.truetype.useMyMetrics`.

**Comment by jenskutilek** [comment]
Looks good to me, I have updated https://github.com/googlefonts/ufo2ft/pull/335 accordingly.

**Pull request merged**
